src/darwin/Framework/CHIP/MTRDeviceControllerDelegateBridge.mm

```diff
@@ -120,14 +121,28 @@
     id<MTRDeviceControllerDelegate> strongDelegate = mDelegate;
     MTRDeviceController * strongController = mController;
     if (strongDelegate && mQueue && strongController) {
-        if ([strongDelegate respondsToSelector:@selector(controller:commissioningComplete:nodeID:)]) {
+        if ([strongDelegate respondsToSelector:@selector(controller:commissioningComplete:nodeID:)] ||
+            [strongDelegate respondsToSelector:@selector(controller:commissioningComplete:nodeID:metrics:)]) {
             dispatch_async(mQueue, ^{
                 NSError * nsError = [MTRError errorForCHIPErrorCode:error];
                 NSNumber * nodeID = nil;
                 if (error == CHIP_NO_ERROR) {
                     nodeID = @(nodeId);
                 }
-                [strongDelegate controller:strongController commissioningComplete:nsError nodeID:nodeID];
+
+                if ([strongDelegate respondsToSelector:@selector(controller:commissioningComplete:nodeID:metrics:)]) {
+                    MTRMetrics * metrics = [MTRMetrics new];
+
+                    if (nsError) {
+                        [metrics setValue:nsError forKey:MTRMetricCommissioningStatusKey];
+                    } else {
+                        auto * error = [NSError errorWithDomain:MTRErrorDomain code:0 userInfo:nil];
```

Comment:
Why this instead of nil?  At the very least, this behavior should be documented somehere in the API.  Because comparing `NSError *` to nil is what people will normally reach for...